test/integration/hds_integration_test.cc

```diff
@@ -0,0 +1,153 @@
+#include "envoy/api/v2/eds.pb.h"
+#include "envoy/api/v2/endpoint/endpoint.pb.h"
+#include "envoy/service/discovery/v2/hds.pb.h"
+
+#include "common/config/resources.h"
+
+#include "test/config/utility.h"
+#include "test/integration/http_integration.h"
+#include "test/test_common/network_utility.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+class HdsIntegrationTest : public HttpIntegrationTest,
+                           public testing::TestWithParam<Network::Address::IpVersion> {
+public:
+  HdsIntegrationTest() : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, GetParam()) {}
+
+  // Used as args to updateClusterLocalityAssignment().
+  struct LocalityAssignment {
+    LocalityAssignment() : LocalityAssignment({}, 0) {}
+    LocalityAssignment(const std::vector<uint32_t>& endpoints) : LocalityAssignment(endpoints, 0) {}
+    LocalityAssignment(const std::vector<uint32_t>& endpoints, uint32_t weight)
+        : endpoints_(endpoints), weight_(weight) {}
+
+    // service_upstream_ indices for endpoints in the cluster.
+    const std::vector<uint32_t> endpoints_;
+    // If non-zero, locality level weighting.
+    const uint32_t weight_{};
+  };
+
+  void createUpstreams() override {
+    fake_upstreams_.emplace_back(new FakeUpstream(0, FakeHttpConnection::Type::HTTP2, version_));
+    hds_upstream_ = fake_upstreams_.back().get();
+    HttpIntegrationTest::createUpstreams();
+  }
+
+  void initialize() override {
+    setUpstreamCount(upstream_endpoints_);
+    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v2::Bootstrap& bootstrap) {
+      // Setup hds and corresponding gRPC cluster.
+      auto* hds_confid = bootstrap.mutable_hds_config();
+      hds_confid->set_api_type(envoy::api::v2::core::ApiConfigSource::GRPC);
+      hds_confid->add_grpc_services()->mutable_envoy_grpc()->set_cluster_name("hds_report");
+      auto* hds_cluster = bootstrap.mutable_static_resources()->add_clusters();
+      hds_cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+      hds_cluster->mutable_circuit_breakers()->Clear();
+      hds_cluster->set_name("hds_report");
+      hds_cluster->mutable_http2_protocol_options();
+      // Put ourselves in a locality that will be used in
+      // updateClusterLoadAssignment()
+      auto* locality = bootstrap.mutable_node()->mutable_locality();
+      locality->set_region("some_region");
+      locality->set_zone("zone_name");
+      locality->set_sub_zone(sub_zone_);
+      // Switch predefined cluster_0 to EDS filesystem sourcing.
```

Comment:
I think we can skip this and the other EDS aspects for now. Granted, we will want to potentially add some of this back later if we do a combined EDS/HDS test, but since we're just doing HDS for now, I think we can skip this bit (and its related state/helpers in this file).